Paintroid/src/main/java/org/catrobat/paintroid/iotasks/OpenRasterFileFormatConversion.java

```diff
@@ -0,0 +1,276 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+//
+//Source:	https://www.openraster.org/baseline/file-layout-spec.html
+//
+//Layout:
+//example.ora  [considered as a folder-like object]
+//        ├ mimetype
+//        ├ stack.xml
+//        ├ data/
+//        │  ├ [image data files referenced by stack.xml, typically layer*.png]
+//        │  └ [other data files, indexed elsewhere]
+//        ├ Thumbnails/
+//        │  └ thumbnail.png
+//        └ mergedimage.png
+public final class OpenRasterFileFormatConversion {
+	private static final String TAG = OpenRasterFileFormatConversion.class.getSimpleName();
+	private static final int COMPRESS_QUALITY = 100;
+	private static final int THUMBNAIL_WIDTH = 256;
+	private static final int THUMBNAIL_HEIGHT = 256;
+
+	private OpenRasterFileFormatConversion() {
+		throw new AssertionError();
+	}
+
+	public static Uri exportToOraFile(List<Bitmap> bitmapList, String fileName, Bitmap bitmapAllLayers, ContentResolver resolver) throws IOException {
+		Uri imageUri;
+		OutputStream outputStream;
+		ContentValues contentValues = new ContentValues();
+		float wholeSize = 0;
+
+		String mimeType = "image/openraster";
+		byte[] mimeByteArray = mimeType.getBytes();
+		byte[] xmlByteArray = getXmlStack(bitmapList);
+
+		for (Bitmap current: bitmapList) {
+			ByteArrayOutputStream bos = new ByteArrayOutputStream();
+			current.compress(Bitmap.CompressFormat.PNG, COMPRESS_QUALITY, bos);
+			byte[] byteArray = bos.toByteArray();
+			wholeSize += byteArray.length;
+		}
+
+		ByteArrayOutputStream bosMerged = new ByteArrayOutputStream();
+		bitmapAllLayers.compress(Bitmap.CompressFormat.PNG, COMPRESS_QUALITY, bosMerged);
+		byte[] bitmapByteArray = bosMerged.toByteArray();
+
+		Bitmap bitmapThumb = Bitmap.createScaledBitmap(bitmapAllLayers, THUMBNAIL_WIDTH, THUMBNAIL_HEIGHT, false);
+		ByteArrayOutputStream bosThumb = new ByteArrayOutputStream();
+		bitmapThumb.compress(Bitmap.CompressFormat.PNG, COMPRESS_QUALITY, bosThumb);
+		byte[] bitmapThumbArray = bosThumb.toByteArray();
+
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+
+			//applefile has no file ending. which is important for api level 30.
+			// we can't save an application file in media directory so we have to save it in downloads.
+			contentValues.put(MediaStore.Files.FileColumns.DISPLAY_NAME, fileName);
+			contentValues.put(MediaStore.Files.FileColumns.RELATIVE_PATH, Environment.DIRECTORY_DOWNLOADS + "/Pocket_Paint");
```

Comment:
```suggestion
			contentValues.put(MediaStore.Files.FileColumns.RELATIVE_PATH, Environment.DIRECTORY_DOWNLOADS);
```
Directly in the downloads folder.